File: src/main/java/j/chain_of_resposibility/like/functional/SpecialSupport.java
package j.chain_of_resposibility.like.functional;

public final class SpecialSupport {

    public static Support create(String name, int number) {
        return t -> {
            if (t.getNumber() == number) {
                MessagePrinter.done(name, t);
                return ControlBreak.of(t).withControlBreak(true);
            } else {
                return ControlBreak.of(t);
            }
        };
    }

}

File: src/main/scala/s/chain_of_resposibility/like/functional/SpecialSupport.scala
package s.chain_of_resposibility.like.functional

import s.chain_of_resposibility.Trouble

object SpecialSupport {

  def create(name: String, number: Int): PartialFunction[Trouble, Unit] = {
    case trouble if trouble.number == number =>
      MessagePrinter.done(name, trouble)
  }
}
